gino/ext/tornado.py

```diff
@@ -320,184 +224,17 @@ class Application(tornado.web.Application):
-                            coro = gen.throw(*exc_info)
-                    except StopIteration:
-                        return
-                    exc_info = None
-                    data = None
-                    # noinspection PyBroadException
-                    try:
-                        data = await coro
-                    except:  # NOQA
-                        exc_info = sys.exc_info()
-            finally:
-                await self._teardown_connection()
-        except Exception as e:
-            # noinspection PyBroadException
-            try:
-                self._handle_request_exception(e)
-            except Exception:
-                tornado.log.app_log.exception("exception in exception handler")
-        finally:
-            if self._prepared_future is not None:
-                if not self._prepared_future.done():
-                    self._prepared_future.set_result(None)
-
-
-# noinspection PyAbstractClass
-class GinoRequestHandler(AsyncioRequestHandler):
-    """
-    Base class for all request handlers that use GINO.
-
-    In addition to features provided by :py:class:`~.AsyncioRequestHandler`,
-    this class manages lazy connections for each request.
-
-    """
-
-    __db_connection = None
-
-    @property
-    def use_connection_for_request(self):
-        """
-        If ``True``, a lazy connection is created for each request.
-
-        That is, whenever the first query occurs, a new connection is
-        borrowed from the application's db object. All succeeding queries made
-        within this request will reuse that connection. The connection will be
-        returned to the pool once the request is finished or the
-        :py:meth:`~.release_connection()` method is called explicitly.
-
-        This property is equal to
-        :py:attr:`Application.use_connection_for_request`
-        by default.
-
-        """
-        return self.application.use_connection_for_request
-
-    @property
-    def db(self):
-        """
-        Access to the database object.
-
-        This property is equal to :py:attr:`Application.db` by default.
-
-        """
-        return self.application.db
-
-    async def _setup_connection(self):
-        if self.use_connection_for_request:
-            self.__db_connection = await self.db.acquire(lazy=True)
-
-    async def _teardown_connection(self):
-        if self.__db_connection is not None:
-            await self.__db_connection.release()
-            self.__db_connection = None
-
-    @property
-    def db_connection(self):
-        """
-        The actual connection associated with this request or ``None`` if
-        ``use_connection_for_request`` is ``False``.
-
-        """
-        return self.__db_connection
-
-    async def release_connection(self):
-        """
-        Return the connection associated with this request back to the pool.
-
-        """
-        await self._teardown_connection()
+        asyncio_loop.run_until_complete(set_bind())
```

Comment:
This also seems a bit wired, why don't we make this whole function `async`?

Reply:
yes, that should be better. I'll fix it.